src/parser/HLSTree.cpp

```diff
@@ -390,9 +390,9 @@ HLSTree::PREPARE_RESULT HLSTree::prepareRepresentation(Period* period,
   if (!rep->source_url_.empty())
   {
     SPINCACHE<Segment> newSegments;
-    unsigned int newStartNumber;
+    uint64_t newStartNumber;
     Segment newInitialization;
-    uint32_t segmentId(rep->getCurrentSegmentNumber());
+    uint64_t segmentId(rep->getCurrentSegmentNumber());
```

Comment:
this change require more attention on line 751:
`rep->current_segment_ = rep->get_segment(segmentId - rep->startNumber_);`
`get_segment` method parameter accept uint32 not uint64
have you check for possible problems due to uint32?

Reply:
I think get_segment should still accept uint32 as there's never going to be billions of segments, in the playlist example in the OP the expression would evaluate as `20220118074272 - 20220118074255` (== 17)  
How do feel about then casting the input for these cases i.e `rep->current_segment_ = rep->get_segment(static_cast<uint32_t>(segmentId - rep->startNumber_));` ? I don't think we'd ever suffer any loss of data but I know it's not good how it currently is.

Reply:
mmh ok then add only a cast to uint32_t in the passed get_segment argument

IMO in another cleanup PR i think `SPINCACHE` type `pos` variable/method should be changed to size_t, since all this struct works with size_t and vector